source/core/Events.cpp

```diff
@@ -548,6 +549,31 @@ static inline EventSource GetEventSourceForEventType(EventType eType) {
     return eSource;
 }
 
+void GetVIName(VirtualInstrument *vi, StringRef viName) {
+    PercentEncodedSubString encodedStr(vi->VIName(), true, false);
+    SubString encodedSubstr = encodedStr.GetSubString();
+    viName->AppendSubString(&encodedSubstr);
+}
+
+void RegisterForControlEvent(EventInfo *eventInfo, EventSpec eventSpec, StringRef viName, EventQueueID qID, UInt32 controlID, RefNum reference,
```

Comment:
Adding another helper for registering the events since it's used in multiple places. For ControlRefNums, this will end up calling jsRegisterForControlEvent. We may be able to do this for JavaScriptRefNum too and avoid having a separate RegisterForJSEvent instruction call.